### PR TITLE
Added phonetic match filter

### PR DESCRIPTION
**Pull request by richplant**
Add a filter that searches for keyword matches using phonetic matching algorithms

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -8,6 +8,9 @@ torch
 num2words==0.5.10
 word2number==1.1
 
+# for PhoneticMatchFilter
+jellyfish==0.8.2
```
Comment (kaustubhdhole) [inline requirements.txt:12]: This should be in a separate requirements.txt in the folder `soundex`

Comment (RyanGoslingsBugle) [reply requirements.txt]: Resolved in 26eed69dba6909b7a04b2d544a3b92834c7b74a2

**Review by tuetschek — COMMENTED** [review]
Looks mostly good to me! The filter is reasonable and the code looks mostly OK, the required jellyfish library is BSD-licensed. 

I'd have a few suggestions though:
* In the readme, I'd somehow separate the example command from the example output
* Wouldn't this be applicable to other tasks as well (e.g. text tagging)? I don't see why not.
* I don't like the default keywords, IMO they will just cause confusion – I'd suggest you remove them.

**Comment by tuetschek** [comment]
@RyanGoslingsBugle Thanks for the updates! Just a question – are you sure this should be applied to *all* task types?

**Comment by RyanGoslingsBugle** [comment]
No prolem. I don't see which tasks it _wouldn't_ be applicable to - even in say, an RDF triple, you may wish to filter subject entities based on their soundalike quality, if you expect typos in your dataset for instance.

**Review by tuetschek — APPROVED** [review]
@RyanGoslingsBugle good point – all good IMO :-)

**Comment by RyanGoslingsBugle** [comment]
@tuetschek cheers!

**Review by kaustubhdhole — APPROVED** [review]

**Pull request merged**